### DISCUSSION TITLE
Right/Left Cycle with screen switching - bug or feature?

### DISCUSSION THREAD

**Discussion by kennethpensopay** [mrkai77/Loop#420] [Q&A]
So I installed Loop today, and I have a 3 monitor setup. So I configured my keybindings for Right and Left Cycle to add a Screen Switch in there to be able to move it to another screen. (Screenshots attached)

But let's say I press TriggerKey + Left; then I would assume that the windows moved to the left in the order I have chosen in the Left cycle. So let's just say left half screen.

But if I then press TriggerKey + Right, I would assume it would do the first thing in my Right Cycle list.
Pressing TriggerKey + Left would then again do the first thing in my Left Cycle. Or at least, that's what I expected.

If I keep pressing TriggerKey + Left, TriggerKey + Right, TriggerKey + Left etc. then the window randomly either move from side to side on one screen, or move around on all my screens.

I think I would prefer that the cycles worked like this:

When Left/right/top/bottom cycle is triggered, it starts with the first position in the cycle list. And if I trigger the same cycle again, it should go to the next one. But if I then trigger another cycle, and then back to the initial cycle, it should start over from the first action in the cycle.

So if the first position in the Left cycle is Left half of the screen, and the second position is left 1/3 of the screen, and I then do a TriggerKey + Right, i expect the window to go to the first position in my Right cycle list (right half of the screen). If I then press TriggerKey + Left again, I expect it to move to the left half of the screen. NOT the left 1/3 or the screen or move to the previous screen.

This is a bit difficult to explain, but I hope that you understand what I mean. :P 

![Screenshot 2024-06-19 at 12 48 20](https://github.com/MrKai77/Loop/assets/144011149/68c9dd2f-6fab-404e-a0f6-8879c82a1725)
![Screenshot 2024-06-19 at 12 48 25](https://github.com/MrKai77/Loop/assets/144011149/827bbc7b-25c9-4a86-a105-1148e14f36f2)

**Comment by JaceThings** [comment]
hmm.. this seems like a bug, as it doesn't happen to me.

Here's my setup:

![000536@2x](https://github.com/MrKai77/Loop/assets/60551230/20748201-4bae-4f92-947b-e5e992c04f89)


When swapping between cycles, the initial cycle is always reset.


https://github.com/MrKai77/Loop/assets/60551230/6fdc8a3e-b42a-40d8-9b5d-ff9c56173ded

**Reply by mrkai77** [reply]
Yes, this seems like a bug. Made an issue for this at #431, will try and fix soon!
